src/mito2/src/engine.rs

```diff
@@ -188,6 +189,29 @@ impl EngineInner {
         Ok(())
     }
 
+    async fn set_readonly_gracefully(&self, region_id: RegionId) -> Result<SetReadonlyResult> {
+        if let Some(region) = self.workers.get_region(region_id) {
+            if !region.is_writable() {
+                let last_entry_id = region.version_control.current().last_entry_id;
```

Comment:
However, if users call `set_writable(false)` directly in other threads, it's possible that the `last_entry_id` will change later as there may be pending writes in the worker.

We might always have to go through the worker unless we don't allow setting readonly in other ways.

Reply:
Good point!

Reply:
I changed it to submitting all requests to the worker.

Reply:
However, it is still unable to handle the corner cases like:

set_readonly_gracefully
write region 1 
set_writable(1, true)
write region 1 

And the first write will fail, but the second write will succeed.

Reply:
Yes. But I think this is not the race condition about `set_readonly_gracefully()`. The entry id returned by `set_readonly_gracefully()` is still correct before the next time the region becomes writable.

However, we might find a better way to unify `set_writable`, `set_readonly`, and region management in the future.